src/middlewared/middlewared/plugins/chart_releases_linux/resources.py

```diff
@@ -15,18 +15,23 @@ class Config:
         namespace = 'chart.release'
 
     @private
-    async def retrieve_pod_with_containers(self, release_name):
+    async def retrieve_pod_with_containers(self, release_name, retrieve_active_pods=False):
         await self.middleware.call('kubernetes.validate_k8s_setup')
         release = await self.middleware.call(
             'chart.release.query', [['id', '=', release_name]], {'get': True, 'extra': {'retrieve_resources': True}}
         )
         choices = {}
-        for pod in release['resources']['pods']:
-            choices[pod['metadata']['name']] = []
+        for pod in filter(
+            lambda p: retrieve_active_pods and p['status']['phase'] == 'Running' or not retrieve_active_pods,
```

Comment:
K8s api can be inconsistent sometimes, are we certain these keys will always be there ?